week4/assignment1/src/pages/search/Search.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import { AxiosError } from 'axios';
+import { searchMembers } from '../../services/memberService';
+import PageContainer from '../../layouts/pageContainer/PageContainer';
+import Title from '../../components/title/Title'
+import Input from '../../components/input/Input'
+import Button from '../../components/button/Button'
+import Header from '../../components/header/Header';
+import * as styles from './Search.css';
+
+
+const Search = () => {
+  const [search, setSearch] = useState('');
+  const [isSearched, setIsSearched] = useState(false);
+  const [nicknameList, setNicknameList] = useState<string[]>([]);
+  const userId = Number(localStorage.getItem('userId'));
+
+  const handleSearch = async () => {
+  try {
+    const res = await searchMembers(userId, search.trim());
+    setNicknameList(res.data.data.nicknameList);
+  } catch (error) {
+    const err = error as AxiosError<{ message: string }>;
+    alert(err.response?.data?.message || '회원 검색에 실패했습니다.');
+  } finally {
+    setIsSearched(true);
+  }
+
+  setSearch('')
+}; 
+
+  return (
+    <>
+        <Header />
+        <PageContainer>
+            <Title>SOPT 회원 조회하기</Title>
+            <Input
+                label="닉네임"
+                type="text"
+                placeholder="검색할 닉네임을 입력하세요"
+                value={search}
+                onChange={(e) => setSearch(e.target.value)}
+            />
+            <Button onClick={handleSearch}>확인</Button>
+
+            {nicknameList.length > 0 && (
+            <ul className={styles.nicknameResult}>
+              {nicknameList.map((name, index) => (
+                  <li key={index}>{name}</li>
+              ))}
+            </ul>
```

Comment:
key값으로 index는 사용하지 않는게 좋습니다!
데이터 순서가 바뀌면 모든 요소가 리렌더링이 발생하게돼요.

Reply:
name은 중복될 수 있어서 key 값으로 사용하지 못했는데, 이런 경우에는 어떤 값을 key로 사용하는 게 좋을까요?!

Reply:
```suggestion
            <ul className={styles.nicknameResult}>
              {nicknameList.map((name, index) => (
                  <li key={name + "-" + index}>{name}</li>
              ))}
            </ul>
```

원래라면 백엔드측에게 고유한 값을 추가해달라고 하겠지만, 지금 같은 상황에서는 이렇게하는것도 대안이 될 것 같아요 ㅎㅎ
@1jiwoo27

Reply:
감사합니다! 수정했습니다 ㅎㅎ